Refactor data input to use 'input' instead of 'inputdata_directory'

moritzpschwarz opened the pull request:
Replaces the 'inputdata_directory' argument with a more flexible 'input' parameter across functions, documentation, and tests. The new 'input' can accept file paths, data.frames, or lists, improving usability and clarity. Updates error handling, test cases, and deprecates 'inputdata_directory' with appropriate warnings.

moritzpschwarz committed:
Refactor data input to use 'input' instead of 'inputdata_directory'

Replaces the 'inputdata_directory' argument with a more flexible 'input' parameter across functions, documentation, and tests. The new 'input' can accept file paths, data.frames, or lists, improving usability and clarity. Updates error handling, test cases, and deprecates 'inputdata_directory' with appropriate warnings.

moritzpschwarz commented on R/load_or_download_variables.R:
```diff
@@ -626,87 +644,120 @@ download_edgar <- function(to_obtain, quiet) {
 
-    if(!quiet){
-      cat("Variable provided as 'inputdata_directory' seems to be a data.frame type. Used as data source.\n")
+  if(is.data.frame(input)){
+    input <- list(input)
+  }
+
+  for(inp in seq_along(input)){
+    cur_input <- input[[inp]]
+    if(is.character(cur_input)){
+      filepaths <- c(filepaths, cur_input)
     }
+    if(is.data.frame(cur_input)){
```
There is no change here in terms of code

moritzpschwarz committed:
Update vignettes to use new 'input' argument in run_model

Replaces references to 'inputdata_directory' with the new 'input' argument in example-workflow.Rmd and new_variable_to_dict.Rmd. Updates documentation and code examples to reflect the new way of providing local data to run_model().

moritzpschwarz committed:
Deprecate inputdata_directory in favor of input

Replaces usage and documentation of the 'inputdata_directory' argument with 'input' throughout the codebase, including function calls, examples, and documentation. Updates run_model to provide a more robust deprecation warning for 'inputdata_directory'. Adjusts README and vignettes to reflect the new argument name and usage.

moritzpschwarz committed:
Merge branch 'main' into input_change

moritzpschwarz committed:
Correct small bug in test-4-data-input

moritzpschwarz commented:
@jkurle any views on this? I think I'm fairly finished with the change.

jkurle commented on R/run_model.R:
```diff
@@ -163,6 +166,12 @@ run_model <- function(specification,
+      stop("'inputdata_directory' is deprecated. Use 'input' instead.")
+    }
+  }
+
```
I think technically, we should make the function still work with the old argument for users who already used it...so if someone points to a directory, we could (should) proceed as before, read the directory, create the list of files or dfs ourselves and give it to input, then warn the user (with the lifecycle package). Depends on how well we want to handle this from a software engineering perspective.

If we don't and simply break all old behaviour, I think we should make a "release" for the previous logic before we merge this PR so that users would be able to install that version easily from GitHub?

jkurle reviewed (commented):
I think the above is my main comment: do we want to handle this gracefully and allow the inputdata_directory to still work as before (for at least a while) or do we want this to be a breaking change? Since it has been mostly us using it (as far as we are aware), it is probably fine to break and we haven't released to CRAN yet anyway. However, then we should probably first make the current main branch as a version release and then introduce this as a version increase with breaking changes.

jkurle commented:
On another note: I was unable to confirm that the branch works as intended. I have unit tests failing and raising warnings, but I also have that on the current main branch. Do all your tests succeed locally? Since most of the tests are turned off on CI or CRAN, I am unsure whether the tests are truly faulty but pass on GH Actions because we skip them; or whether my environment setup is somehow causing problems.

moritzpschwarz commented:
Alright, will implement - simply detect if it is not a (list of) file(s), then pass with a warning that this was deprecated and might be removed. 
Before merging create a release from main.

moritzpschwarz committed:
Fix lag usage and update test expectations

Replaces base R lag() with dplyr::lag() in test scripts for consistency and to avoid ambiguity. Updates expected values in super exogeneity tests to match new results. Also removes an unused variable from test-4-data-input.R.

moritzpschwarz committed:
Improve handling of deprecated inputdata_directory argument

Enhanced the run_model function to retain support for the deprecated 'inputdata_directory' argument with improved error and warning messages. Added logic to handle both directory and data frame inputs, and updated tests to cover correct and incorrect usage, ensuring warnings are issued and errors are thrown as appropriate.

moritzpschwarz committed:
Attempting to repair test-3

moritzpschwarz committed:
Merge branch 'main' into input_change

jkurle committed:
update test with expected behaviour

jkurle reviewed (approved):
looks all good to me now!
